File: src/main/java/org/noxet/noxetserver/messaging/channels/MessagingPartyChannel.java
package org.noxet.noxetserver.messaging.channels;

import org.bukkit.command.CommandSender;
import org.noxet.noxetserver.minigames.party.Party;

import java.util.HashSet;
import java.util.Set;

public class MessagingPartyChannel extends MessagingChannel {
    private final Party party;

    public MessagingPartyChannel(Party party) {
        this.party = party;
    }

    @Override
    public Set<CommandSender> getRecipients() {
        return new HashSet<>(party.getMembers());
    }
}

File: src/main/java/org/noxet/noxetserver/messaging/channels/MessagingPartyChannel.scala
package org.noxet.noxetserver.messaging.channels

import org.bukkit.command.CommandSender
import org.noxet.noxetserver.minigames.party.Party

import scala.collection.Set
import scala.collection.immutable.HashSet

class MessagingPartyChannel(party: Party) extends MessagingChannel:
    override def getRecipients: Set[CommandSender] = party.getMembers